### DISCUSSION TITLE
Small UX, logging, and internal code improvements. Some questions

### DISCUSSION THREAD

**Discussion by RollingStar** [beetbox/beets#4933] [Ideas]
I didn't realize this post would be so big. I will snip these out into PRs if people like them.

## Elaborate on "no files imported" when "something" happened.

Simple fix!

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L1397

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L1172

```
log.warning('No files imported from {0} ({1} skipped)',
                        displayable_path(toppath), task_factory.skipped)
```

## Log archives

Can a file go down the code path for `is_archive` (true or false) and not end up in `skipped` or `imported`? If so we should count them and log them in the above log. I want a folder `/myimports/` where I just dump new music in and don't think about it. Currently beets requires single archive imports, so you can't do it in bulk. If we [can't support](https://github.com/beetbox/beets/discussions/4932) a folder of archives, I'd like to see a log that there are X files in the folder but we didn't import any of them.

## Log to console when not all archives are supported.

Would it make sense to proactively test archive support?

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L1105

It's pretty quick to install the 7z and rar support. It might be good to have a note at the start of beets -v that RAR and/or 7z are not supported. It should link to a docs page on enabling them. Or maybe just log in-place when the file is attempted to be opened? It would get crowded on the log, but again, it's easy to install 7z and rar and quiet that warning.

## Include 7z and rar in the recommended packages for beets on pip

Not familiar with how python packaging works but can/should we do this? How is an end user supposed to interact with this? https://github.com/beetbox/beets/blob/master/setup.py

## New log when quiet.

```
Evaluating 19 candidates.

Sending event: import_task_before_choice
Importing as-is.
Sending event: import_task_choice
found duplicates: [2]
default action for duplicates: a
--->This album is already in the library!
--->Skipping.
Sending event: import
Sending event: cli_exit
```

Maybe combine the two messages?

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/ui/commands.py#L860

`Album is already in the library! Quiet enabled, skipping duplicate resolution.`

That is 78 characters (77 for "Item") which is below 79 characters which I assume is our target max line width for the console? Combining the messages would help the user understand that they both apply to the same item IMO.

## New log for listing duplicates

The log currently lists the internal beet album ID. (see `beet ls beatles -f '$albumid'`). 

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L1526

```
log.debug('found duplicates: {}'.format(
                [o.id for o in found_duplicates]
```

This should at least say: `Found duplicates with IDs: `. I had a dupe with ID 2 and I thought that there were 2 duplicates. Can we log the duplicate_keys that are identical as well? Ex.

```
Found duplicates with duplicate_keys [albumartist album].

Album: The Beatles - Revolver. Duplicate IDs are: 2, 4, 159, 300
```

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L1540

## Resolve_duplicates

How does this work? I have it set to `a`sk and beets works fine. The code looks like it should raise NotImplementedError because `a` is not in the if-else. I'm guessing the class is [extended somewhere](https://stackoverflow.com/questions/60392220/what-is-notimplementederror)?

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L1540

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L302

## Fix the docstring on find_duplicates

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L681

The dupe logic uses `duplicate_keys` from the config. the docstring is outdated.

## Remove outdated check for empty artist/asis

https://github.com/beetbox/beets/blob/e10b955a931e4c205b0cadf0860797c0aeee736c/beets/importer.py#L688

```
    if info['artist'] is None:
            # As-is import with no artist. Skip check.
            return []
```

This would appear to be unnecessary now that the user can define their own dupe keys. If it's kept, it would need to be dynamic in response to the duplicate_keys.  The comment notes "As-is import with no artist. Skip check." but I don't feel like tracking down how this function is called. If you do as-is import and don't have an artist, what should happen?

**Comment by JOJ0** [comment]
I like

New log for listing duplicates
